### PR TITLE
add new formula for skim

### PR DESCRIPTION
**Pull request by lotabout**
Skim is a fuzzy finder written in rust. Kind of [fzf](https://github.com/junegunn/fzf) replacement if you know what I'm talking about.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by lotabout** [comment]
@BrewTestBot test this please

File: Formula/skim.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def caveats; <<~EOS
+    To install useful keybindings and fuzzy completion:
+      #{opt_prefix}/install
+    To use skim in Vim, add the following line to your .vimrc:
+      set rtp+=#{opt_prefix}
+  EOS
+  end
+
+  test do
+    system "#{bin}/sk", "--version"
```
Comment (fxcoudert) [inline Formula/skim.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/skim.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  sha256 "41280bee2138afefff95f76a640b753d1cb9215e8391ef37eab7ccb3517d9019"
+
+  depends_on "rust" => :build
+
+  def install
+    (buildpath/"src/github.com/lotabout").mkpath
+    ln_s buildpath, buildpath/"src/github.com/lotabout/skim"
+    system "cargo", "install", "--root", prefix, "--path", "."
+
+    prefix.install "install"
```
Comment (fxcoudert) [inline Formula/skim.rb]: `prefix/"install"` is not a standard location

File: Formula/skim.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    (prefix/"shell").install %w[bash zsh].map { |s| "shell/key-bindings.#{s}" }
```
Comment (fxcoudert) [inline Formula/skim.rb]: Neither is `prefix/"shell"`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/skim.rb (hunk 1 above)
Comment (lotabout) [reply Formula/skim.rb]: @fxcoudert `skim` is an interactive tool, does brew have some way to simulate user inputs?

File: Formula/skim.rb (hunk 3 above)
Comment (lotabout) [reply Formula/skim.rb]: @fxcoudert I actually followed fzf's formula: https://github.com/Homebrew/homebrew-core/blob/master/Formula/fzf.rb

Could you please tell me what's the standard location for storing shell bindings and vim plugins?

Comment (igas) [reply Formula/skim.rb]: https://github.com/Homebrew/homebrew-core/blob/8d0d13e425ea70d42d9597cd90e38e8949150329/Formula/interactive-rebase-tool.rb#L21-L49

File: Formula/skim.rb (hunk 1 above)
Comment (igas) [reply Formula/skim.rb]: https://github.com/Homebrew/homebrew-core/blob/8d0d13e425ea70d42d9597cd90e38e8949150329/Formula/interactive-rebase-tool.rb#L21-L49

**Comment by jab** [comment]
Thanks for submitting this, @lotabout! Was hoping to try skim. Sorry to see this got auto-closed before it landed. Are you interested in picking this back up?

**Comment by lotabout** [comment]
@jab I will pick it up later.

Brew formula need to contain some tests while I could not make it work since skim needs user interaction. Later I'll add filter mode so that skim could be invoked silently.

**Pull request closed without merging**